aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_bulletproofs.move

```diff
@@ -99,12 +106,69 @@ module aptos_std::ristretto255_bulletproofs {
         )
     }
 
+    /// Verifies a zero-knowledge range proof for a batch of Pedersen commitments `comms`, ensuring that all values
+    /// `v` satisfy `v` in `[0, 2^num_bits)`.
+    /// Only works for `num_bits` in `{8, 16, 32, 64}` and batch size (length of `comms`) in `{1, 2, 4, 8, 16}`.
+    public fun verify_batch_range_proof_pedersen(
+        comms: &vector<pedersen::Commitment>, proof: &RangeProof,
+        num_bits: u64, dst: vector<u8>): bool
+    {
+        assert!(features::bulletproofs_batch_enabled(), error::invalid_state(E_NATIVE_FUN_NOT_AVAILABLE));
+
+        let comms = std::vector::map_ref(comms, |com| ristretto255::point_to_bytes(&pedersen::commitment_as_compressed_point(com)));
+
+        verify_batch_range_proof_internal(
+            comms,
+            &ristretto255::basepoint(), &ristretto255::hash_to_point_base(),
+            proof.bytes, num_bits, dst
+        )
+    }
+
+    // Verifies a zero-knowledge range proof for a batch of commitments `comms` (each of the form 
+    /// `v * val_base + r * rand_base`), ensuring that all values `v` satisfy 
+    /// `v` in `[0, 2^num_bits)`. Only works for `num_bits` in `{8, 16, 32, 64}` and batch size 
+    /// (length of the `comms`) in `{1, 2, 4, 8, 16}`.
+    public fun verify_batch_range_proof(
+        comms: &vector<pedersen::Commitment>,
```

Comment:
This is supposed to take a `vector<RistrettoPoint>`, no?

Reply:
The other one takes a vector of Pedersen commitments.

Reply:
Fixed